### ISSUE TITLE
Relationships without links cause problems

### ISSUE THREAD

**Issue by ryyppy** [confetti/yayson#17]
Hello again!
I was trying to wrap my head around the `store.js` code after hitting some kind of edge-case and I don't know if this might be a bug.

So I tried following code (the JSON-API data was copied straight from the response but I removed unnecessary stuff):

```
//Example 1 (failing)
store.sync({
    "data": {
        "id": "1",
        "type": "event",
        "attributes": {
            "name": "Hello World Event"
        },
        "relationships": {
            "author": {
                "data": {
                    "type": "author",
                    "id": "5"
                }
            },
            "comments": {
                "data": null
            },
            "prev": {
                "data": null
            },
            "next": {
                "data": null
            }
        }
    }
});

//Causes following exception:
TypeError: Cannot read property 'links' of null
    at Store.toModel (node_modules/yayson/lib/yayson/store.js:48:35)
    at Store.sync (node_modules/yayson/lib/yayson/store.js:160:21)
    at lib/strategy/http.js:136:12
    at lib$rsvp$$internal$$tryCatch (node_modules/rsvp/dist/rsvp.js:493:16)
    at lib$rsvp$$internal$$invokeCallback (node_modules/rsvp/dist/rsvp.js:505:17)
    at lib$rsvp$$internal$$publish (node_modules/rsvp/dist/rsvp.js:476:11)
    at lib$rsvp$asap$$flush (node_modules/rsvp/dist/rsvp.js:1198:9)

//(The error happens in coffee-script line store.js:37)
```

In example 1, I expected the `sync()` call to store the record without any resolved `author` attribute (only `{type: '...', id: '...'`).

For comparison, after I try to sync the same dataset with an `included` field, the operation works as expected:

```
//Example 2 (working)
store.sync({
    "data": {
        "id": "1",
        "type": "event",
        "attributes": {
            "name": "Hello World Event"
        },
        "relationships": {
            "author": {
                "data": {
                    "type": "author",
                    "id": "5"
                }
            },
            "comments": {
                "data": null
            },
            "prev": {
                "data": null
            },
            "next": {
                "data": null
            }
        }
    },
    "included": [
        {
            "id": "5",
            "type": "author",
            "attributes": {
                "firstname": "John",
                "lastname": "Doe"
            }
        }
    ]
});

store.find('event', '1');

//Returned model:
{
    "id": "1",
    "name": "Hello World Event",
    "author": {
        "id": "5",
        "get": [Function],
        "links": {}
    }
}

```

Maybe I misunderstood the concept of the returned `Model` literal, since I don't understand why there is a requirement for links to exist (it is optional in the specification).

Do I actually require `links` for unresolved relationship attributes?

Again for comparison, example 2 does not require any links in any resource for the method to work.

If I found a funny edge case, then I will try to fix this and make a pull request, but first I need to know if I am on the right path and not overlooking something  ;-)

Thanks!
Pat

**Comment by ajcobo** [comment]
I just tried your example and it definitely fails when attributes are not provided. Reading the JSONAPI documentation it clearly does not require to attach additional data in the relationship.

I think you are in the right direction. 

The tests on store does not include this case. You could add the test and provide the PR.

If you need help in testing or solving the situation I can help you with that for sure.

**Comment by ryyppy** [comment]
Ah cool, I will try to fix that and create the PR the next few days.
Thanks for your feedback @ajcobo !

**Comment by jede** [comment]
Yes, sound like a bug. A PR would be great!

**Comment by rubencosta** [comment]
@jede bump :)

**Comment by jede** [comment]
Oh, sorry! Missed this, I'll take a look!

**Comment by Staff457** [comment]
**Dr.Agbalazzy, you are indeed a miracle worker. I have tried so many spell casters and psychic after my husband left me and told me to go and see the attorney for the divorce. And Dr.Agbalazzy was the only spell caster that could guarantee me an urgent 24hours spell casting.. and he assured me that my husband is coming back.. I trusted Dr. Agbalazzy, and its was more than a dream to see my husband back to my life with lots of apologies and love on the 24th of january 2015, And he was remorseful for all his wrong actions, For sure this is really a miracle temple. Olorunoduduwaspiritualtemple@gmail.com**
